cachi2/core/package_managers/rpm/redhat.py

```diff
@@ -0,0 +1,168 @@
+import asyncio
+import hashlib
+import logging
+import uuid
+from collections import UserDict
+from os import PathLike
+from pathlib import Path
+from typing import Any, Optional, Union
+
+from pydantic import BaseModel, PositiveInt, ValidationError
+
+from cachi2.core.config import get_config
+from cachi2.core.package_managers.general import async_download_files
+
+log = logging.getLogger(__name__)
+
+
+class Package(BaseModel):
+    """Package item; represents RPM or SRPM file. Used for parsing the lockfile."""
+
+    repoid: Optional[str] = None
+    url: str
+    checksum: Optional[str] = None
+    size: Optional[int] = None
+
+
+class Arch(BaseModel):
+    """Architecture structure. Used for parsing the lockfile."""
+
+    arch: str
+    packages: list[Package] = []
+    source: list[Package] = []
+
+
+class Root(BaseModel):
+    """Top of the structure of the lockfile. Used for parsing the lockfile."""
+
+    lockfileVersion: PositiveInt
+    lockfileVendor: str
+    arches: list[Arch]
+
+
+class Header(BaseModel):
+    """
+    The class represents header of the lockfile. Used for matching the format.
+
+    Contains just a minimal set of fields to identify the format.
+    """
+
+    lockfileVersion: PositiveInt
+    lockfileVendor: str
+
+
+class RedhatRpmsLock(UserDict):
+    """
+    The class implements basic operations with specific lockfile format.
+
+    Input data comes from raw yaml stored as a dictionary.
+    """
+
+    # parsed structure of the lockfile
+    _lockfile: Root
+    # metadata of file records found in the lockfile (for verification and SBOM generation)
+    _files_metadata: dict[Path, Any] = {}
+    # 'repoid' key is not mandatory in the lockfile format. When not present,
+    # fallback is set as (partly) random string containing _uuid
+    _uuid = uuid.uuid4().hex[:6]
+
+    @property
+    def internal_repoid(self) -> str:
+        """Internal_repoid getter."""
+        return f"cachi2-{self._uuid}"
+
+    @property
+    def internal_source_repoid(self) -> str:
+        """Internal_source_repoid getter."""
+        return f"cachi2-{self._uuid}-source"
+
+    def is_internal_repoid(self, repoid: str) -> bool:
+        """Repoid directory matches the internal repoid."""
+        return repoid in (self.internal_repoid, self.internal_source_repoid)
```

Comment:
None of these properties ought to be public accessors IMO. The first 2 should be converted to "private" methods (possibly static if you  won't keep uuid as an attribute) and the last one can be dropped as it only should be needed in `inject_files` and we can handle the check there.

Reply:
> and we can handle the check there.
Do you mean not to pass the lockfile object anymore and not to use `is_intetnal_repoid`? And recognize an internal repoid from its specific format?

Reply:
Yes.

Reply:
OK, you answered it elsewhere:
https://github.com/containerbuildsystem/cachi2/pull/502#discussion_r1551770780

Reply:
_internal_repoid_ / _internal_source_repoid_ are currently used in the `download` method. You recommended to move `download` out from `RedhatRpmsLock`. And then, I would have to call these (private) methods outside of object. I would speak for keeping them public.